mitodl/micromasters
# Added unique constraint to CourseRun edx_course_key

#### What are the relevant tickets?

Fixes #1409 

#### What's this PR do?

Adds a unique constraint to CourseRun.edx_course_key

#### How should this be manually tested?

The following should cause an error in a django shell:

```python
from courses.models import Program, Course, CourseRun
edx_key = '12345'
course = Course.objects.first()
new_course_run = CourseRun.objects.create(
  course=course,
  edx_course_key=edx_key
)
CourseRun.objects.create(
  course=course,
  edx_course_key=edx_key
)
new_course_run.delete()
```

The second to last line should cause an integrity error

#### Any background context you want to provide?

~~Not sure if we're going to have an issue with this `edx_course_key` being nullable. Only one run would be able to have a null value. Should this field be nullable anyway?~~ <-- this shouldn't be a problem (see comment below), but the reviewer should make sure to test this
